Lib/test/test_traceback.py

```diff
@@ -1060,6 +1060,21 @@ def test_exception_group_deep_recursion_traceback(self):
         self.assertIn('ExceptionGroup', output)
         self.assertLessEqual(output.count('ExceptionGroup'), LIMIT)
 
+    @cpython_only
+    def test_print_exception_bad_type_capi(self):
+        from _testcapi import exception_print
+        with captured_output("stderr") as stderr:
+            exception_print(42)
+        self.assertEqual(
+            stderr.getvalue(),
+            ('TypeError: print_exception(): '
+             'Exception expected for value, int found\n')
+        )
+
+    def test_print_exception_bad_type_python(self):
+        with self.assertRaises(TypeError):
```

Comment:
Is there a chance other `TypeErrors` may sneak in here? Would it make sense to also match the exception message using `assertRaisesRegex`?